apps/storybook/src/TiledHeatmapMesh/checkerboard-api.ts

```diff
@@ -0,0 +1,53 @@
+import type { Size } from '@h5web/lib';
+import { getLayerSizes, TilesApi } from '@h5web/lib';
+import greenlet from 'greenlet';
+import { clamp } from 'lodash';
+import type { NdArray } from 'ndarray';
+import ndarray from 'ndarray';
+import { createFetchStore } from 'react-suspense-fetch';
+import type { Vector2 } from 'three';
+
+import type { TileParams } from './models';
+import { areTilesEqual } from './utils';
+
+const getCheckerboardArray = greenlet(
+  async (length: number, value: number): Promise<Uint8Array> => {
+    return Uint8Array.from({ length }, () => value);
+  }
+);
+
+export class CheckerboardTilesApi extends TilesApi {
+  private readonly store;
+
+  public constructor(size: Size, tileSize: Size) {
+    super(tileSize, getLayerSizes(size, tileSize));
+
+    this.store = createFetchStore(
```

Comment:
Using a suspense store like in the mandlebrot API in order for the `get` method to remain synchronous.